File: 0283-move-zeroes/0283-move-zeroes.cpp
// TC :- O(N), SC :- O(1) 

class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int n = nums.size(), count = 0;
        for(int i = 0; i < n; i++) {
            if(nums[i] != 0) nums[count++] = nums[i]; // storing non-zero elements first
        }
        while(count < n) nums[count++] = 0; // storing zeroes at the end
    }
};

File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int n = nums.length, count = 0;
        for(int num : nums) {
            if(num != 0) {
                nums[count++] = num;
            }
        }

        while(count < n) {
            nums[count++] = 0;
        }
    }
}